### PR TITLE
Updating capabilities dictionary, requirements.txt and black CI check

### PR DESCRIPTION
**Pull request by antalszava**
* Updates the capabilities dictionary as per the changes in PennyLane core (https://github.com/PennyLaneAI/pennylane/pull/781).
* Add missing requirements to the requirements.txt file and adjust checks.
* Updating `black` formatting CI check

### CODE REVIEW

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -70,6 +70,8 @@ jobs:
 
       - name: Install cibuildwheel
         run: |
+          python -m pip install --upgrade pip
+          pip install git+https://github.com/PennyLaneAI/pennylane.git
```
Comment (antalszava) [inline .github/workflows/wheels.yml:74]: Will be removed

File: pennylane_lightning/lightning_qubit.py [hunk 2]
```diff
@@ -94,9 +93,22 @@ def __init__(self, wires, *, shots=1000, analytic=True):
 
         if self.num_wires > self._MAX_WIRES:
             warnings.warn(
-                "The number of wires exceeds 16, reverting to NumPy-based evaluation.", UserWarning,
+                "The number of wires exceeds 16, reverting to NumPy-based evaluation.",
+                UserWarning,
```
Comment (antalszava) [inline pennylane_lightning/lightning_qubit.py:97]: `black`

File: setup.py [hunk 3]
```diff
@@ -168,7 +168,8 @@ def build_extensions(self):
 
 requirements = [
     "numpy",
-    "pennylane>=0.11.0",
+    "pennylane @ git+https://github.com/PennyLaneAI/pennylane.git",
```
Comment (antalszava) [inline setup.py:171]: Will be reverted

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -54,7 +54,10 @@ jobs:
 
       - name: Get required Python packages
         run: |
-          pip install numpy pybind11 pytest pytest-cov pytest-mock flaky
+          cd main
+          python -m pip install --upgrade pip
+          pip install git+https://github.com/PennyLaneAI/pennylane.git
+          pip install -r requirements.txt
```
Comment (antalszava) [inline .github/workflows/build.yml:60]: was discovered by Maria, that we were not depending on `requirements.txt` here

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -54,7 +54,10 @@ jobs:
 
       - name: Get required Python packages
         run: |
-          pip install numpy pybind11 pytest pytest-cov pytest-mock flaky
+          cd main
+          python -m pip install --upgrade pip
```
Comment (antalszava) [inline .github/workflows/build.yml:58]: Added `pip` upgrades just for making sure all is good...

**Review by trbromley — APPROVED** [review]
Thanks @antalszava!

File: requirements.txt [hunk 6]
```diff
@@ -1,5 +1,7 @@
+flaky
 numpy
 pennylane>=0.11
```
[the remark this answers is not in the document]
Comment (antalszava) [reply requirements.txt:3]: Yep, in the version bump PR. Note, that that PR should pass checks only when PL v0.12 has been released.

**Pull request merged**